modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationTasks.java

```diff
@@ -8,47 +8,61 @@
 import static com.opengamma.strata.collect.Guavate.toImmutableList;
 
 import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.stream.IntStream;
+import java.util.Set;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.light.LightMetaBean;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
 import com.opengamma.strata.basics.CalculationTarget;
 import com.opengamma.strata.basics.market.ReferenceData;
 import com.opengamma.strata.calc.CalculationRules;
+import com.opengamma.strata.calc.CalculationRunner;
 import com.opengamma.strata.calc.Column;
-import com.opengamma.strata.calc.config.FunctionConfig;
-import com.opengamma.strata.calc.config.Measure;
-import com.opengamma.strata.calc.config.ReportingCurrency;
-import com.opengamma.strata.calc.config.pricing.ConfiguredFunctionGroup;
-import com.opengamma.strata.calc.config.pricing.FunctionGroup;
 import com.opengamma.strata.calc.marketdata.MarketDataRequirements;
 import com.opengamma.strata.calc.marketdata.MarketDataRequirementsBuilder;
-import com.opengamma.strata.calc.marketdata.mapping.MarketDataMappings;
+import com.opengamma.strata.calc.runner.function.CalculationFunction;
 import com.opengamma.strata.collect.Messages;
 
 /**
  * The tasks that will be used to perform the calculations.
+ * <p>
+ * This captures the targets, columns and tasks that define the result grid.
+ * Each task can be executed to produce the result. Applications will typically
+ * use {@link CalculationRunner} or {@link CalculationTaskRunner} to execute the tasks.
  */
-public final class CalculationTasks {
+@BeanDefinition(style = "light")
+public final class CalculationTasks implements ImmutableBean {
 
   /**
-   * The targets.
+   * The targets that will be calculated.
+   * <p>
+   * The result of the calculations will be a grid where each row is taken from this list.
    */
+  @PropertyDefinition(validate = "notEmpty")
   private final List<CalculationTarget> targets;
   /**
-   * The columns.
+   * The columns that will be calculated.
+   * <p>
+   * The result of the calculations will be a grid where each column is taken from this list.
    */
+  @PropertyDefinition(validate = "notEmpty")
   private final List<Column> columns;
   /**
-   * The calculation tasks.
+   * The tasks that perform the individual calculations.
    * <p>
-   * The tasks in the list are arranged in row order.
-   * For example, if a grid has 5 columns, the calculations 0-4 in the list are the first row,
-   * calculations 5-9 are the second row and so on.
+   * The results can be visualized as a grid of columns with a row for each target.
+   * There is one calculation for each cell in the grid. The calculations in the list are arranged in row order.
```

Comment:
This comment doesn't really apply to this list any more as a task can span multiple columns.